cmd/bpf2go/main.go

```diff
@@ -161,40 +202,27 @@ func run(stdout io.Writer, pkg, outputDir string, args []string) (err error) {
 		return fmt.Errorf("-output-stem %q must not contain path separation characters", b2g.outputStem)
 	}
 
-	targetArches := strings.Split(*flagTarget, ",")
-	if len(targetArches) == 0 {
-		return fmt.Errorf("no targets specified")
-	}
-
-	targets, err := collectTargets(targetArches)
+	targetArches, err := collectTargets(strings.Split(*flagTarget, ","))
 	if errors.Is(err, errInvalidTarget) {
-		printTargets(stdout)
-		fmt.Fprintln(stdout)
+		printTargets(b2g.stdout)
```

Comment:
Maybe move these into the caller in a separate `errors.Is(err, errInvalidTarget)` branch? Then we don't print at all in this function as far as I can tell.